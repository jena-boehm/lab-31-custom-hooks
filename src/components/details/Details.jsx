import React from 'react';
import PropTypes from 'prop-types';
import styles from '../app/styles/App.css';

const Details = ({ name, temperament, lifeSpan, origin, bredFor, image }) => {
  if(origin) return (
    <>
      <figure>
        <figcaption className={styles.nameHeader}>{name}</figcaption>
        <img src={image} alt={name} width="600px" />
      </figure>
      <div className={styles.details}>
        <span className={styles.detailLabel}>Temperament: </span>
        <span>{temperament}</span><br/>
        <span className={styles.detailLabel}>Life Span: </span>
        <span>{lifeSpan}</span><br/>
        <span className={styles.detailLabel}>Origin: </span>
        <span>{origin}</span><br/>
        <span className={styles.detailLabel}>Bred For: </span>
        <span>{bredFor}</span>
      </div>
    </>
  );

  else return (
    <>
      <figure>
        <figcaption className={styles.nameHeader}>{name}</figcaption>
        <img src={image} alt={name} width="600px" />
      </figure>
      <div className={styles.details}>
        <span className={styles.detailLabel}>Temperament: </span>
        <span>{temperament}</span><br/>
        <span className={styles.detailLabel}>Life Span: </span>
        <span>{lifeSpan}</span><br/>
        <span className={styles.detailLabel}>Bred For: </span>
        <span>{bredFor}</span>
      </div>
    </>
  );
};

Details.propTypes = {
  name: PropTypes.string,
  temperament: PropTypes.string,
  lifeSpan: PropTypes.string,
  origin: PropTypes.string,
  bredFor: PropTypes.string,
  image: PropTypes.string,
};

export default Details;
