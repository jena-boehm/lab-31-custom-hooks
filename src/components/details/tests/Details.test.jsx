import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Details from '../Details';
import dogByName from '../../../fixtures/dogByName.json';

describe('Details component', () => {
  afterEach(() => cleanup());
  it('renders Details', () => {
    const { asFragment } = render(<Details component={[dogByName]}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
