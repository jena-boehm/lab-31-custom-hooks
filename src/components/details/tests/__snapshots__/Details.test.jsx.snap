// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<DocumentFragment>
  <figure>
    <figcaption />
    <img
      width="400px"
    />
    <p>
      Temperament: 
    </p>
    <p>
      Life Span: 
    </p>
    <p>
      Origin: 
    </p>
  </figure>
</DocumentFragment>
`;
